api/views.py

```diff
@@ -56,6 +56,26 @@ def tasks_list(request):
         )
     return JsonResponse({"tasks": result})
 
+def tasks_list_all(request: HttpRequest):
```

Comment:
This view should be teacher-only, so please mark it with `@user_passes_test(is_teacher)`. It would also be great if we could unify it with `tasks_list` (we just need to make sure that it returns data in a format that is compatible with existing users of `tasks_list`, and make it a bit more REST-friendly, by moving the subject parameter into the URL, i.e. `/tasks/<subject>` (where the subject would be optional).

Reply:
I don't know, if url `/tasks/subject` will be possible, since there is already path `/tasks/\<path to task\>` which show details about task: https://github.com/mrlvsb/kelvin/blob/master/api/urls.py#L9

Reply:
Uhh, the paths are a mess :) Then use something else, e.g. `/task-list/...`, or something like that